### ISSUE TITLE
TSRMLS_D not declare .

### ISSUE THREAD

**Issue by lpflpf** [CopernicaMarketingSoftware/PHP-CPP#274]
**env:**
       gcc version 4.8.2 20131212
       PHP 7.0.1 (cli) (built: Jan  7 2016 09:52:33) ( **ZTS** DEBUG )  (install from php source code.)
       php-config in global path.

**Question:**
    
 ```
mkdir -p shared/common
mkdir -p shared/zend
g++ -Wall -c -std=c++11 -fvisibility=hidden -DBUILDING_PHPCPP -Wno-write-strings -MD -g `php-config --includes` -fpic -o shared/zend/base.o zend/base.cpp
In file included from /usr/local/php701/include/php/Zend/zend_portability.h:52:0,
                 from /usr/local/php701/include/php/Zend/zend_types.h:27,
                 from /usr/local/php701/include/php/Zend/zend.h:31,
                 from /usr/local/php701/include/php/main/php.h:35,
                 from zend/includes.h:41,
                 from zend/base.cpp:8:
zend/origexception.h:39:5: error: declaration does not declare anything [-fpermissive]
     TSRMLS_D;
     ^
In file included from zend/includes.h:129:0,
                 from zend/base.cpp:8:
zend/origexception.h: In constructor ‘Php::OrigException::OrigException(zend_object*)’:
zend/origexception.h:51:24: error: expected unqualified-id before ‘=’ token
         this->TSRMLS_C = TSRMLS_C;
                        ^
zend/origexception.h:51:34: error: expected primary-expression before ‘;’ token
         this->TSRMLS_C = TSRMLS_C;
                                  ^
zend/origexception.h: In copy constructor ‘Php::OrigException::OrigException(const Php::OrigException&)’:
zend/origexception.h:64:18: error: expected primary-expression before ‘=’ token
         TSRMLS_C = exception.TSRMLS_C;
                  ^
zend/origexception.h:64:38: error: expected unqualified-id before ‘;’ token
         TSRMLS_C = exception.TSRMLS_C;
                                      ^
zend/origexception.h: In constructor ‘Php::OrigException::OrigException(Php::OrigException&&)’:
zend/origexception.h:80:18: error: expected primary-expression before ‘=’ token
         TSRMLS_C = exception.TSRMLS_C;
                  ^
zend/origexception.h:80:38: error: expected unqualified-id before ‘;’ token
         TSRMLS_C = exception.TSRMLS_C;
                                      ^
make: *** [shared/zend/base.o] Error 1

```

**TSRMLS_D** can be find the declared in **TSRM.h**:

```
/usr/local/php701/include/php/TSRM/TSRM.h:183:#define TSRMLS_D void
/usr/local/php701/include/php/TSRM/TSRM.h:184:#define TSRMLS_DC
/usr/local/php701/include/php/TSRM/TSRM.h:208:#define TSRMLS_D	void
/usr/local/php701/include/php/TSRM/TSRM.h:209:#define TSRMLS_DC
```

**Comment by edwardstock** [comment]
@lpflpf look at this issue https://github.com/CopernicaMarketingSoftware/PHP-CPP/issues/266

**Comment by martijnotto** [comment]
I believe this is fixed in commit #b4ccea9
